### PR TITLE
Gizmo challenge

### PR DESCRIPTION
**Pull request by enniin**
This is my first PR for this challenge.

### CODE REVIEW

**Comment by enniin** [comment]
Yay!

File: gizmo.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+"""
+
+import numpy as np
+import pandas as pd
+
+
+# -------------------- Exercise 1 --------------------
+
+def hello(name, country='Finland'):
+    print('Hello ' + name + ', how are things in ' + country + '?')
```
Comment (wmvanvliet) [inline gizmo.py:18]: Alternatively, you can use Python f-strings:

```python
print(f'Hello {name}, how are things in {country}?')
```

File: gizmo.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    print(out[:-1])
+
+# -------------------- Exercise 3 --------------------
+
+def relative_path(ids):
+    base = ['./subjects/mock_recording_','.rec']
+    names = []
+    
+    if type(ids) == str:
+        ids = [ids]
```
Comment (wmvanvliet) [inline gizmo.py:35]: Love the attention to detail!

File: gizmo.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    for s in ids:
+        names.append(base[0] + s + base[1])
```
Comment (wmvanvliet) [inline gizmo.py:37]: F-Strings are super userful here as well:
```python
names.append(f'./subjects/mock_recording_{s}.rec')
```

File: gizmo.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+def multiplication_table(zero_out_multiples=None):
+    """Times table with optional dropping of multiples
+    
+    Returns the times table for integers up to 12, and optionally
+    zeroes the multiples of the given int argument.
+    
+    Parameters
+    ----------
+    zom : int, optional
```
Comment (wmvanvliet) [inline gizmo.py:61]:
```suggestion
    zero_out_multiples : int, optional
```

File: gizmo.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    Returns
+    -------
+    tab : numpy array
+        The times table as an array
+        
+    """
+
+    v = np.arange(1,13)
+    tab = np.outer(v,v)
+    if type(zero_out_multiples) == int:
```
Comment (wmvanvliet) [inline gizmo.py:73]: Interesting way to check that `zero_out_multiples` is not `None`. Personally I recommend:

```python
if zero_out_multiples is not None:
```

File: gizmo.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        first = second
+        second = third
+
+# ------------------- Exercise 12 --------------------
+
+def get_fibonacci_sequence(n):
+    fib = generate_fibonacci_sequence(n)
+    arr = np.empty(0)
+    for f in fib:
+        arr = np.append(arr,f)
```
Comment (wmvanvliet) [inline gizmo.py:94]: `np.append` is super slow. I recommend: `np.array(list(generate_fibonacci_sequence(n)))`

File: gizmo.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+def get_titanic():
+    df = pd.read_csv('titanic.csv')
+    return df
+
+# ------------------- Exercise 14 --------------------
+
+def get_titanic_children(age=12):
+    df = get_titanic()
+    return df.query('age <= @age')
```
Comment (wmvanvliet) [inline gizmo.py:107]: I learned something new from your code! I dind't know about `@`. Cool!

**Review by wmvanvliet — APPROVED** [review]
Great job!

File: gizmo.py (hunk 1 above)
Comment (enniin) [reply gizmo.py:18]: Awesome, thanks for the tip!

File: gizmo.py (hunk 4 above)
Comment (enniin) [reply gizmo.py:61]: True! Forgot about the docstring there...

File: gizmo.py (hunk 5 above)
Comment (enniin) [reply gizmo.py:73]: Ahh, true! But I kinda also wanted to catch floats etc.

File: gizmo.py (hunk 6 above)
Comment (enniin) [reply gizmo.py:94]: VERY good tip. I seriously need to wean myself from np.append... Didn't realise that list() can be used like that!

File: gizmo.py (hunk 7 above)
Comment (enniin) [reply gizmo.py:107]: It also works like 'var == @array(@ind]' and all sorts of fancy stuff! Been using it a lot recently. Prob not super efficient (I mean, using pandas and query in general), but I feel it makes the code easily readable – at least if you think in data bases and sql, which I guess I do. 😆

File: gizmo.py (hunk 5 above)
Comment (wmvanvliet) [reply gizmo.py:73]: But a float should probably raise an error, not be silently ignored

**Pull request closed without merging**